### PR TITLE
Use $doCheck instead of $scope.$watch in filter, sort, and toolbar co…

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
…mponents

### CODE REVIEW

File: src/filters/filter-fields-component.js [hunk 1]
```diff
@@ -40,42 +41,49 @@ angular.module('patternfly.filters').component('pfFilterFields', {
...
-        ctrl.config.currentValue = undefined;
+        ctrl.addFilterFn(ctrl.currentField, ctrl.currentValue);
+        ctrl.currentValue = undefined;
       }
     }
 
     function setupConfig () {
-      if (ctrl.fields === undefined) {
-        ctrl.fields = [];
+      prevConfig = ctrl.config;
```
Comment (dtaylor113) [inline src/filters/filter-fields-component.js]: I think you need to copy the current ctrl.config, else prevConfig will always equal ctrl.config as ctrl.config gets updated.  Ie.   `prevConfig = angular.copy(ctrl.config);`

File: src/filters/filter-results-component.js [hunk 2]
```diff
@@ -40,13 +41,20 @@ angular.module('patternfly.filters').component('pfFilterResults', {
...
+    ctrl.$doCheck = function () {
+      // do a deep compare on config
+      if (!angular.equals(ctrl.config, prevConfig)) {
+//        setupConfig();
+      }
     };
 
     function setupConfig () {
+      prevConfig = ctrl.config;
+
```
Comment (dtaylor113) [inline src/filters/filter-results-component.js]: angular.copy()

File: src/sort/sort-component.js [hunk 3]
```diff
@@ -3,31 +3,36 @@ angular.module('patternfly.sort').component('pfSort', {
...
+      if (!angular.equals(ctrl.config, prevConfig)) {
         setupConfig();
-      }, true);
+      }
     };
 
     function setupConfig () {
       var updated = false;
 
+      prevConfig = ctrl.config;
```
Comment (dtaylor113) [inline src/sort/sort-component.js]: angular.copy()

File: src/toolbars/toolbar-component.js [hunk 4]
```diff
@@ -21,18 +22,29 @@ angular.module('patternfly.toolbars').component('pfToolbar', {
...
-      }, true);
+    ctrl.$doCheck = function () {
+      // do a deep compare on config
+      if (!angular.equals(ctrl.config, prevConfig)) {
+        setupConfig();
+      }
     };
 
+    function setupConfig () {
+      prevConfig = ctrl.config;
```
Comment (dtaylor113) [inline src/toolbars/toolbar-component.js]: angular.copy()

**Comment by jeff-phillips-18** [comment]
Changed to angular.copy, thanks @dtaylor113

**Pull request merged**